### PR TITLE
Adding more browser capabilities

### PR DESCRIPTION
**Pull request by birojnayak**
- [y ] You've read the [Contributor Guide](../CONTRIBUTING.md) and [Code of Conduct](../CODE-OF-CONDUCT.md).
- [ n/A] You've included unit tests for your change, where applicable.
- [ n/a] You've included inline docs for your change, where applicable.
- [ y] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

**PR Title**
Adding more browser capabilities

**PR Description**
Detail 1
Detail 2

Addresses #bugnumber (in this specific format)

### CODE REVIEW

File: src/Microsoft.AspNetCore.SystemWebAdapters/Configuration/HttpCapabilitiesBase.cs [hunk 1]
```diff
@@ -53,6 +53,14 @@ private IHttpBrowserCapabilityFeature Capability
 
     public string? this[string key] => Capability[key];
 
+    public Version? EcmaScriptVersion => new Version(Capability["ecmascriptversion"]);
+
+    public Version? MSDomVersion => new Version(Capability["msdomversion"]);
+
+    public Version? W3CDomVersion => new Version(Capability["w3cdomversion"]);
+
+    public virtual bool SupportsCallback => GetBoolean("supportsCallback");
```
Comment (twsouthwick) [inline src/Microsoft.AspNetCore.SystemWebAdapters/Configuration/HttpCapabilitiesBase.cs:62]: let's remove the virtual here

File: src/Microsoft.AspNetCore.SystemWebAdapters/Configuration/HttpCapabilitiesBase.cs [hunk 2]
```diff
@@ -53,6 +53,14 @@ private IHttpBrowserCapabilityFeature Capability
 
     public string? this[string key] => Capability[key];
 
+    public Version? EcmaScriptVersion => new Version(Capability["ecmascriptversion"] ?? string.Empty);
```
Comment (twsouthwick) [inline src/Microsoft.AspNetCore.SystemWebAdapters/Configuration/HttpCapabilitiesBase.cs:56]:
```suggestion
    public Version? EcmaScriptVersion =>Capability["ecmascriptversion"] ?? is {} v ? new Version(v) : null;
```

You'll probably want to make a helper method to do that. However, verify what .NET Framework did here - did it return a null (I'm assuming that based on your nullable annotation), or did it return a version of something?

**Review by twsouthwick — CHANGES_REQUESTED** [review]

File: src/Microsoft.AspNetCore.SystemWebAdapters/Configuration/HttpCapabilitiesBase.cs (hunk 2 above)
Comment (birojnayak) [reply src/Microsoft.AspNetCore.SystemWebAdapters/Configuration/HttpCapabilitiesBase.cs:56]: https://referencesource.microsoft.com/#mscorlib/system/version.cs,354  and here https://referencesource.microsoft.com/#mscorlib/system/version.cs,376 <= eventually if the version is null or empty or invalid it throws exception.  So I am making the API as not nullable to stick with framework and adding helper to handle as below. Let me know if looks below, I will submit the next revision.

```c#
    public Version EcmaScriptVersion => GetVersion("ecmascriptversion");
   
    private Version GetVersion(string key)
    {
        var result = Capability[key];
        if (result is not null)
        {
            return new Version(result);
        }
        else
        {
            throw new ArgumentNullException(nameof(key));
        }
    }
```

Comment (twsouthwick) [reply src/Microsoft.AspNetCore.SystemWebAdapters/Configuration/HttpCapabilitiesBase.cs:56]: If that's what framework is doing, that sounds good to me

**Review by twsouthwick — APPROVED** [review]

**Pull request merged**